File: 1200. Minimum Absolute Difference/1200.go
package leetcode

import(
	"sort"
)

func minimumAbsDifference(arr []int) [][]int {
	sort.Ints(arr)
	absolute := 100000000
	for i:=0; i<len(arr)-1; i++{
		v := arr[i+1]-arr[i]
		if v<absolute{
			absolute = v
			if absolute == 1{
				break
			}
		}
	}

	relist := make([][]int, 0)
	for i:=0; i<len(arr)-1; i++{
		if arr[i+1]-arr[i] == absolute{
			relist = append(relist, []int{arr[i], arr[i+1]})
		}
	}
	return relist
}

File: 1200. Minimum Absolute Difference/_1200.py
class Solution:
    def minimumAbsDifference(self, arr):
        arr.sort()
        absolute = float("inf")
        for i in range(len(arr)-1):
            absolute = min(absolute, arr[i+1]-arr[i])
            if absolute == 1:
                break

        relist = []
        for i in range(len(arr)-1):
            if arr[i+1] == arr[i]+absolute:
                relist.append([arr[i], arr[i+1]])
        return relist

    # 很python的代码
    def minimumAbsDifference1(self, arr):
        arr.sort()
        mn = min(b - a for a, b in zip(arr, arr[1:]))
        return [[a, b] for a, b in zip(arr, arr[1:]) if b - a == mn]  

